bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/handler/SpeedTestHandler.java

```diff
@@ -132,7 +130,17 @@ public void onError(final @Nullable SpeedTestError testError, final @Nullable St
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMessage);
             freeRefreshTask();
             return;
-        } else if (SpeedTestError.SOCKET_TIMEOUT.equals(testError) || SpeedTestError.SOCKET_ERROR.equals(testError)
+        } else if (SpeedTestError.SOCKET_TIMEOUT.equals(testError)) {
+            timeouts--;
+            if (timeouts <= 0) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Max timeout count reached");
```

Comment:
How can the user reset this counter? I'm thinking of e.g. a network connection problem which takes half a day or so. 

Or maybe a setting of `configuration.maxTimeouts=0` disables timeouts at all?

Reply:
Disabling the thing is an option no ?

Reply:
Sounds good. But what do you think of the timeout-counter disable?